downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -358,6 +358,11 @@ abstract class VersionControlSystem {
 
         addMetadataRevision(pkg.vcsProcessed.revision)
 
+        if (type == VcsType.GIT && revisionCandidates.singleOrNull() == "master") {
```

Comment:
I think this should only be done if "master" is coming from `pkg.vcsProcessed.revision`, because if `listRemoteBranches()` returns only "master" it does not make sense to check "main" as well.